File: Ruby/season2/p027-032/p029.rb
# -*- coding: utf-8 -*-
=begin
　2017.5.2
　while文とuntil文
=end

def main
  i = 1;
  while i < 2000 do
    p i
    i *= 2
  end

  3.times { puts "-----------------------------" }

  i = 1
  until i > 20000 do
    p i
    i *= 3
  end

end


main

File: Python/season2/p029-030/p029.py
# coding: UTF-8
"""
　2018.4.27
　再帰的マージソートの準備
"""

# 統合部分
def merge(left, right):
  if len(left)==0 or len(right)==0:
    return left + right
  else:
    if left[0] < right[0]:
      return [left[0]] + merge(left[1:], right)
    else:
      return [right[0]] + merge(left, right[1:])
    
print(merge([1,3,5], [2,9,11]))